File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/objects/ObjectGroundListAddedMessage.scala





















// Generated on 05/03/2013 10:34:45
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ObjectGroundListAddedMessage extends MessageDeserializer {
  val messageId = 5925

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ObjectGroundListAddedMessage.messageId

    val cells: Seq[Short]
    val referenceIds: Seq[Int]

    def serialize(buf: Buffer) {
      buf.writeUShort(cells.length)
      for (entry <- cells) {
        buf.writeShort(entry)
      }
      buf.writeUShort(referenceIds.length)
      for (entry <- referenceIds) {
        buf.writeInt(entry)
      }
    }

  }

  protected def create(cells: Seq[Short], referenceIds: Seq[Int]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val cells = {
      val builder = collection.immutable.Seq.newBuilder[Short]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readShort()
      }
      builder.result()
    }
    val referenceIds = {
      val builder = collection.immutable.Seq.newBuilder[Int]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readInt()
      }
      builder.result()
    }

    create(cells, referenceIds)
  }

}

object ObjectGroundListAddedMessage extends ObjectGroundListAddedMessage {
  case class MessageImpl(cells: Seq[Short], referenceIds: Seq[Int]) extends Message
  type Target = MessageImpl

  override def create(cells: Seq[Short], referenceIds: Seq[Int]) = MessageImpl(cells, referenceIds)

  def apply(cells: Seq[Short], referenceIds: Seq[Int]) = create(cells, referenceIds)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ObjectGroundListAddedMessage.java


// Generated on 05/08/2013 19:37:47
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ObjectGroundListAddedMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5925;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short[] cells;
    public int[] referenceIds;
    
    public ObjectGroundListAddedMessage() { }
    
    public ObjectGroundListAddedMessage(short[] cells, int[] referenceIds) {
        this.cells = cells;
        this.referenceIds = referenceIds;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeUShort(cells.length);
        for (short entry : cells) {
            buf.writeShort(entry);
        }
        buf.writeUShort(referenceIds.length);
        for (int entry : referenceIds) {
            buf.writeInt(entry);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        int limit = buf.readUShort();
        cells = new short[limit];
        for (int i = 0; i < limit; i++) {
            cells[i] = buf.readShort();
        }
        limit = buf.readUShort();
        referenceIds = new int[limit];
        for (int i = 0; i < limit; i++) {
            referenceIds[i] = buf.readInt();
        }
    }
    
}
